contracts/koh/contracts/KingAutomaton.sol

```diff
@@ -258,10 +286,55 @@ contract KingAutomaton {
     return (p.votes[index] & mask) >> offset;
   }
 
-  function getVoteCount(uint256 ballot_box_id, uint256 _choice) public view validBallotBoxID(ballot_box_id) returns(uint256) {
+  function getVoteCount(uint256 ballot_box_id, uint256 _choice) public view
+      validBallotBoxID(ballot_box_id) returns(uint256) {
     return ballot_boxes[ballot_box_id].voteCount[_choice] & ALL_BUT_MSB;
   }
 
+  function completeProposal(uint256 id) private activeBallot(id) {
+    Proposal storage p = proposals[id];
+    require(p.state == ProposalState.Started || p.state == ProposalState.Contested, "Invalid proposal state!");
+    BallotBox storage b = ballot_boxes[id];
+    p.state = ProposalState.Completed;
+    b.state = BallotBoxState.Inactive;
+  }
+
+  function updateProposalState(uint256 id) public validBallotBoxID(id) {
```

Comment:
This function only needs to check 3 proposal states:
- Started
- Accepted
- Contested

Let's split the logic in 3 sections (simulate a switch with if/else logic) and get rid of the requires.